src/docs/arch-hs.dox

```diff
@@ -1,8 +1,126 @@
 /*! @arch_page arch-hs History Store
 
-The History Store in WiredTiger tracks old (all but the latest committed)
-versions of records.  By having these records in storage separate from
-the current version, they can be used to service long running transactions,
+The history store in WiredTiger tracks historical versions of records required to
+service older readers. By having these records in storage separate from
+the current version, they can be used to service long running transactions
 and be evicted as necessary, without interfering with activity that uses
-the most recent committed versions.
+the most recent committed versions. With the introduction of history store, only the
+newest update to a key is written to the user table while the older updates
+for the key are written to the history store. All user tables in the database are
+backed by a single history store table. The history store has no outward
+visibility to the application code and only WiredTiger's internal modules can
+perform operations on the history store table using a predefined cursor API.
+
+@section arch_hs_table History store table structure
+WiredTiger writes the history store on the disk as a standard row-store table. The key for
+the history store table is a combination of:
+- \c btree ID of user table this update belongs to
+- \c record key (byte-string for row-store, record number for column-store)
+- \c start timestamp for the update
+- \c counter value
+This key format allows us to search efficiently for a given record key and read
+timestamp combination. The last part of the key is a monotonically increasing counter
+to keep the key unique in scenarios where a key has multiple updates with the same commit timestamp.
+As the key for the history store table is different for row- and column-store,
+we store both key types in a byte string otherwise we'd need two history store files with
+different key formats.
+
+The corresponding value for each key is stored as a combination of:
+- \c stop timestamp
+- \c durable timestamp
+- \c update type
+- \c value.
+
+The stop timestamp refers to the point in time after which the update is no longer visible. This can
+either be a result of deletion or of a new update being committed. Every update
+in the history store has a valid stop timestamp and transaction id associated with it, except for
+cases where the update preceded a prepared update. The stop timestamp of the update immediately
+before the prepared update is set to the maximum possible timestamp (\c WT_TS_MAX). The stop
+timestamp of the latest update in the history store is updated once the prepared update is resolved.
+
+@subsection arch_hs_table_tombstone Tombstones in the history store table
```

Comment:
With the previous paragraph, this section seems repetitive. I still think we should explain the stop timestamp is introduced for garbage collection.

Reply:
I have combined the discussion about tombstones in to one section. Have a look at 1fb693b.